PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ServiceBusReceiveModeCheck.java

```diff
@@ -0,0 +1,234 @@
+
+            OptionalInt prefetchCountValue = OptionalInt.empty();
+            boolean isReceiveModePeekLock = false;
+            PsiElement prefetchCountMethod = null;
+
+            // Iterating up the chain of method calls
+            PsiExpression qualifier = methodCall.getMethodExpression().getQualifierExpression();
+
+            // Check if the method call chain has the method to check
+            while (qualifier instanceof PsiMethodCallExpression) {
+
+                // Get the method expression
+                PsiReferenceExpression methodExpression = ((PsiMethodCallExpression) qualifier).getMethodExpression();
+
+                // Get the method name
+                String methodName = methodExpression.getReferenceName();
+
+                // Check if the method name is the method to check
+                if (!(RULE_CONFIG.getMethodsToCheck().contains(methodName))) {
+                    return;
+                }
+                if ("receiveMode".equals(methodName)) {
+                    isReceiveModePeekLock = receiveModePeekLockCheck((PsiMethodCallExpression) qualifier);
+                } else if ("prefetchCount".equals(methodName)) {
+                    prefetchCountValue = getPrefetchCount((PsiMethodCallExpression) qualifier);
+                    prefetchCountMethod = ((PsiMethodCallExpression) qualifier).getMethodExpression().getReferenceNameElement();
+                }
+
+                // Get the qualifier of the method call expression -- the next method call in the chain
+                qualifier = ((PsiMethodCallExpression) qualifier).getMethodExpression().getQualifierExpression();
+
+                // If the receive mode is set to PEEK_LOCK and the prefetch count is set to a value greater than 1, register a problem
+                if (prefetchCountValue.isPresent() && prefetchCountValue.getAsInt() > 1 && isReceiveModePeekLock && prefetchCountMethod != null) {
+                    holder.registerProblem(prefetchCountMethod, RULE_CONFIG.getAntiPatternMessage());
+                    return;
+                }
+            }
+        }
+
+        /**
+         * This method is used to check if the receive mode is set to PEEK_LOCK.
+         *
+         * @param qualifier The method call expression to check
+         * @return true if the receive mode is set to PEEK_LOCK, false otherwise
+         */
+        private boolean receiveModePeekLockCheck(PsiMethodCallExpression qualifier) {
+
+            String peekLockArgument = "PEEK_LOCK";
+
+            // Get the arguments of the method call expression
+            PsiExpression[] arguments = qualifier.getArgumentList().getExpressions();
+
+            for (PsiExpression argument : arguments) {
+                String argumentText = argument.getText();
+
+                // Check if the argument is set to PEEK_LOCK
+                if (argumentText.contains(peekLockArgument)) {
```

Comment:
You used a string literal in lines 165 and 167. I think it's fine to just use `argumentText.contains("PEEK_LOCK")`.